contentcuration/contentcuration/serializers.py

```diff
@@ -68,14 +68,24 @@ def update(self, instance, validated_data):
                     # create new nodes
                     ret.append(File.objects.create(**item))
 
+        files_to_delete = []
+        nodes_to_parse = []
+        current_files = [f['id'] for f in validated_data]
         for file_obj in validated_data:
-            contentnode = file_obj['contentnode']
-            preset = file_obj['preset_id']
-            file_id = file_obj['id']
-            language = file_obj.get('language_id')
-            files_to_delete = File.objects.filter(Q(contentnode=contentnode) & (Q(preset_id=preset) | Q(preset=None)) & (Q(language_id=language)) & ~Q(id=file_id))
-            for to_delete in files_to_delete:
-                to_delete.delete()
+            delete_queryset = File.objects.filter(Q(contentnode=file_obj['contentnode']) & (Q(preset_id=file_obj['preset_id']) | Q(preset=None)) & Q(language_id=file_obj.get('language_id')) & ~Q(id=file_obj['id']))
+            files_to_delete += [f for f in delete_queryset.all()]
```

Comment:
Not sure how many items this is likely to be, but a big list of models can use a lot of memory.

It's also possible to build up a queryset using `|` between two querysets, and then apply an operation to the whole  thing (which often translates to a single SQL statement behind the scenes) but not sure how well that'd work here.